editor/src/components/canvas/remix/remix-utils.tsx

```diff
@@ -383,3 +387,32 @@ export function getRoutesAndModulesFromManifest(
     routingTable,
   }
 }
+
+export function getRouteComponentNameForOutlet(
+  path: ElementPath,
+  metadata: ElementInstanceMetadataMap,
+  projectContents: ProjectContentTreeRoot,
+  pathTrees: ElementPathTrees,
+): string | null {
+  if (!MetadataUtils.isProbablyRemixOutlet(metadata, path)) {
+    return null
+  }
+
+  const outletChildren = MetadataUtils.getImmediateChildrenPathsOrdered(metadata, pathTrees, path)
+  if (outletChildren.length == 0) {
+    return null
+  }
+
+  const uidsToFilePath = getAllUniqueUids(projectContents).uidsToFilePaths
+  const filePath = uidsToFilePath[EP.toUid(outletChildren[0])]
```

Comment:
Minor: You could use `safeIndex` the array lookup and replace the length check above with a null check on the result.